### ISSUE TITLE
[FeatureRequest] Disable the highlight in some removal cases

### ISSUE THREAD

**Issue by aj3423** [aileot/emission.nvim#12]
Hi,

In some cases, the highling is redundant. For example:
- `dd`
   You will only use `dd` when you are sure that the line should be deleted, and the highlighting will cover the next line for 300ms, which makes it more difficult to delete multiple lines repeatedly with `dd`.
- Using `x` or `d` to delete selected text in visual mode
  If the text block is already selected and highlighted, no need to highlight it again after the deletion. 
- `db`, `dw`, ...
  Similar reason.

Is it possible to not show the highlighting in such cases? 

Thanks.

**Comment by aileot** [comment]
Thank you for the try, and opening issue :)

Here are some plans for your suggestion:

- [x] Add `min_row_offset` option to filter as well as the `min_byte` option.

  - It also affects other operations like `:substitute`.

- [x] Add a function to temporarily disable emission.

  - To invoke it on `ModeChanged` with a timer to disable emission in a few milliseconds.
  - Or, to map it to keys. (the last idea to be adopted)

- [ ] Implement another extmark strategy not to overlay texts for removed line emissions.

  - Harder implementation. Once I gave it up for the release, but I will try it again.

Give me a time to think them further before. I will add some of them during this week.

Please let me know if you find other troublesome cases than those in Operator-pending mode.
Any other suggestions are also welcome!

P.S.
At present, please increase `min_byte` until the features are implememted.

**Comment by aj3423** [comment]
Thanks for the quick reply and sharing the idea, I'll be using the `min_byte` in the meantime.

> To invoke it on ModeChanged with a timer to disable emission in a few milliseconds.

Just out of curiosity, why is this "a few milliseconds" for? Would it be easier that just disable the highlighting when it's in visual mode, such as: line-visual, column-visual, or any visual mode? 
(pls never mind this if it doesn't make sense, I know nothing about nvim plugin development)

**Comment by aileot** [comment]
Good point!

I had assumed `ModeChanged` event would not help, since `vim.api.nvim_get_mode()` in the `filter` option cannot detect Visual mode.
However, it would. So, we might not need "a few milliseconds" as you point out :)

I'm having a hard time designing the interface options...

**Comment by aileot** [comment]
Hi, 

Now, emission.nvim v1.1.0 is released!

I believe your suggestions can be fulfilled with the release.
If you are still using my plugin, please try the code snippets in the README section:
https://github.com/aileot/emission.nvim?tab=readme-ov-file#modify-options-on-autocmd-events

Of course, let me know if you still have any issues :)

Thanks.

**Comment by aj3423** [comment]
I just upgraded and tried this `ModeChanged`, but it shows error at every launch. 
![image](https://github.com/user-attachments/assets/301b682d-1adb-4100-b075-298619fa3eb9)

**Comment by aileot** [comment]
Thank you for the reporting!!!

I've missed the simplest test to setup without any opts. Please try v1.1.1!

Sorry in advance I don't have much time to fix any further issues to respond right away unlike this simpler issue.

**Comment by aj3423** [comment]
Thanks for the quick fix. The red error is fixed, it doesn't highlight in visual mode anymore. But the `dd` is still highlighted, I've set `min_row_offset = 1`
```
			require("emission").setup({
				removed = {
					enabled = true,
					min_row_offset = 1,
				},
				on_events = {
				...
```
But it doesn't work.

**Comment by aileot** [comment]
Umm, it's reproduced.

Sorry for the documentation inconsistency, but thank you very much for the report!

I'm trying, but I'm not sure if I could fix it during the rest minutes...

**Comment by aileot** [comment]
Should fixed in v1.1.2. Please try it.
Thanks again for your reporting :)

**Comment by aj3423** [comment]
It's fixed, now it's perfect, thank you!
